### PR TITLE
date time modal time selection fix

### PR DESCRIPTION
**Pull request by kand**
Date time selection modal time changes broke date display.

This branch preserves date display when choosing a new time and adds tests for the controller.

### CODE REVIEW

**Comment by kand** [comment]
@spra85 @MelizzaP @collin @daytonn

**Comment by spra85** [comment]
@kand, can you detail exactly what the bug was that this fixes?

**Comment by kand** [comment]
@spra85 updated comment, it's kind of hard to describe so if it doesn't make sense I can show you the issue in person.

File: app/components/date-time-modal/date-time-modal-controller.js [hunk 1]
```diff
@@ -20,34 +34,47 @@ angular.module('bulbs.cms.dateTimeModal.controller', [
...
-      $scope.TIMEZONE_LABEL = moment.tz(CmsConfig.getTimezoneName()).format('z');
+      $scope.tempTime = angular.copy($scope.tempDatetime);
 
       var timeNowWithOffset = function () {
         return moment.tz(CmsConfig.getTimezoneName());
       };
 
-      // callback function for using datetime calendar because it doesn't work at all in a sensible way
+      // callback function for using datetime calendar because it doesn't work
+      //  at all in a sensible way
```
Comment (daytonn) [inline app/components/date-time-modal/date-time-modal-controller.js]: This comment should detail the non-sensible way in which you are working around, as is it offers no explanation.

File: app/components/date-time-modal/date-time-modal-controller.js [hunk 2]
```diff
@@ -20,34 +34,47 @@ angular.module('bulbs.cms.dateTimeModal.controller', [
...
+        $scope.tempDatetime = ($scope.tempDatetime || moment())
+          .year(newDateAsMoment.year())
+          .month(newDateAsMoment.month())
+          .date(newDateAsMoment.date());
       };
 
       $scope.setDateToday = function () {
         var now = timeNowWithOffset();
-        $scope.tempDatetime = moment().year(now.year()).month(now.month()).date(now.date());
+        $scope.tempDatetime = ($scope.tempDatetime ||  moment())
```
Comment (spra85) [inline app/components/date-time-modal/date-time-modal-controller.js]: I feel like `setDate`, `setDateToday`, and `setDateTomorrow` read so similarly, that maybe they should be combined into one function?

**Comment by spra85** [comment]
Seems fine, other than maybe combining those methods for some DRYness

**Comment by kand** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;Lk205vLn&#x2F;606-webtech-problem-with-promotions"> WEBTECH - problem with promotions</a>

**Pull request merged**